dotnet/src/Agents/Core/AgentGroupChat.cs

```diff
@@ -145,4 +172,17 @@ public AgentGroupChat(params Agent[] agents)
         this._agents = new(agents);
         this._agentIds = new(this._agents.Select(a => a.Id));
     }
+
+    private void EnsureStrategyLoggerAssignment()
+    {
+        if (NullLogger.Instance == this.ExecutionSettings.SelectionStrategy.Logger)
+        {
+            this.ExecutionSettings.SelectionStrategy.Logger = this.LoggerFactory.CreateLogger(this.ExecutionSettings.SelectionStrategy.GetType());
+        }
+
+        if (NullLogger.Instance == this.ExecutionSettings.SelectionStrategy.Logger)
+        {
+            this.ExecutionSettings.TerminationStrategy.Logger = this.LoggerFactory.CreateLogger(this.ExecutionSettings.TerminationStrategy.GetType());
+        }
```

Comment:
1. It feels like `NullLogger.Instance` should be second operand here. We usually write `if (something == null)` and not `if (null == something)`.
2. Both conditions are the same. In line 183, there should be a check for `this.ExecutionSettings.TerminationStrategy.Logger`.
3. It feels like logger should be configured in all cases, even without checking if it's `NullLogger` or not.

Reply:
Its an common C++ convention.  I'll switch if its confusing.

It felt to me that re-invoking the factory may be sub-optimal perf as this will be most likely inside a loop.

Reply:
Updated condition order

Reply:
> Both conditions are the same. In line 183, there should be a check for this.ExecutionSettings.TerminationStrategy.Logger.

This point is still unresolved:
![image](https://github.com/microsoft/semantic-kernel/assets/13853051/950bc73e-3d34-4d12-bbde-50cbf2f2a79f)

Reply:
Copy paste typo.

Reply:
Updated